org.jacoco.core/src/org/jacoco/core/internal/analysis/MethodCoverageCalculator.java

```diff
@@ -130,9 +131,10 @@ private void applyReplacements() {
 			}
 
 			final AbstractInsnNode node = entry.getKey();
-			instructions.put(node, instructions.get(node)
+			replacedInstructions.put(node, instructions.get(node)
 					.replaceBranches(branches, fromInstructions, fromBranches));
```

Comment:
Execution of new tests of `MethodAnalyzerTest` in CI with different JDK versions caught the need to store replaced instructions separately from originals: in this test computation of new statuses for two different instructions uses their original statuses - without separation result depends on the order of entries in `replacements` `Map`.

Food for thought: is it possible to demonstrate similar problem on old code? ie that order of replacements matters.

Also it would be nice to add a check at least into the validation tests that no two filters replace the same instruction.